Cleanse/Subcomponent.swift

```diff
@@ -9,10 +9,13 @@
 import Foundation
 
 
-public protocol Subcomponent : _BaseComponent {
-    /// This is the binding required to construct a new subcomponent
-    associatedtype Seed = Void
-
-    /// Subcomponents can have custom scopes associated with them.
+public protocol Component : _BaseComponent {
+    ///  can have custom scopes associated with them.
     associatedtype Scope = _Unscoped
-}
\ No newline at end of file
+}
+
+
+/// Subcomponents have been renamed to Components
+@available(*, deprecated, renamed="Component")
+public typealias Subcomponent = Component
```

Comment:
Yeah, I think so. Its going to be odd when peeps upgrade from master to this regardless since Component -> RootComponent & Subcomponent -> Component. 

I guess this is why we're a beta version.